web/source/osk/embedded/singleInputCapture.ts

```diff
@@ -0,0 +1,39 @@
+namespace com.keyman.osk.embedded {
+  export class SingleInputCapture {
+    private readonly handler: (event: TouchEvent) => boolean;
+
+    private static readonly eventOptionConfig: AddEventListenerOptions = {
+      capture: true,
+      // While other
+      once: true,
+      passive: false,
+      // While `abort` exists, it's too new to rely upon it for some of the older platforms we support.
+
+    };
+
+    constructor(captureCallback: () => void) {
+      this.handler = (event) => {
+        captureCallback();
+        event.stopPropagation();
+        event.stopImmediatePropagation();
```

Comment:
If I'm understanding this correctly, this is going to block the first keystroke and dismiss the globe hint. That doesn't seem ideal.

Reply:
You are understanding this correctly.  If naught else, I did want a little discussion here.

What would a user's intent most likely be if they tapped the hint bubble?  If a user would want to type a key behind the hint, the hint _is_ pretty much in the way of such a keystroke.  My thought:  they're probably trying to dismiss it, without typing anything.

From that, if we want to swallow the keystroke in _that_ scenario, would it be best to be consistent (UX-wise) and always swallow the first keystroke every time regardless of position?  While not super-ideal for those who know what they're doing, this behavior should be super-easy for users to intuit, understand, and quickly deal with.

Finally, note that hardware keystrokes are not swallowed by this class.

------

I opted for this approach at first because I wanted to be _sure_ to dismiss without a keystroke if the user tapped the help bubble.  But, since the help bubble _is_ stacked over a key that _also_ has event listeners, the question of how to guarantee prevention of the key event arose, and I wasn't super-clear on that since the help bubble is in a separate branch of the DOM tree from the `VisualKeyboard`.

I just did some extra research, and I think I understand how to do bubble-only input-swallowing if that's what we'd prefer now.  Is that what we'd prefer?  Or would we prefer to trigger keys, even if behind the help bubble, at the same time as bubble dismissal?  (That said, while I haven't tested this... I think this last point - the "even if" scenario - may actually be more difficult / require a bigger change!)

Reply:
> ... I did want a little discussion here.
>
>What would a user's intent most likely be if they tapped the hint bubble? If a user would want to type a key behind the hint, the hint is pretty much in the way of such a keystroke. My thought: they're probably trying to dismiss it, without typing anything.

Given that the hint bubble blocks several of the keys, I'm in favor of just dismissing the bubble.

If they press the globe key, should we dismiss the bubble and handle the globe key?

Reply:
From my perspective, the behaviour should be:

1. Any touch should dismiss the bubble
2. Touch on bubble itself should have no further action
3. Touch on globe key should still do globe key action
4. Touch elsewhere should do the action of the key that is touched.

The general principle is that touching somewhere on the screen should interact with the element under the user's finger. If we wanted to disable the entire keyboard, perhaps shading it down, then it would make sense to capture the touch anywhere on the keyboard to dismiss the hint bubble (and remove the shading, marking the keyboard 'ready for use'). But I don't think we want to do that -- it's better to only swallow the touch if it's directly on the bubble itself.

Reply:
This should be in place now.  Feel free to double-check the instructions for the user tests to verify that they properly model the behavior you expect.